mtp_api/apps/prison/tests/test_views.py

```diff
@@ -0,0 +1,124 @@
+
+    def test_cannot_create_if_not_logged_in(self):
+        response = self.client.post(
+            self.list_url, format='json'
+        )
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_create(self):
+        user = self.users[0]
+
+        # create one PrisonerLocation so that we test the upload_counter
+        # increment
+        mommy.make(PrisonerLocation, upload_counter=1)
+        self.assertEqual(PrisonerLocation.objects.count(), 1)
+
+        self.client.force_authenticate(user=user)
+
+        data = [
+            {
+                'prisoner_number': random_prisoner_number(),
+                'prisoner_dob': random_prisoner_dob(),
+                'prison': self.prisons[0].pk
+            },
+            {
+                'prisoner_number': random_prisoner_number(),
+                'prisoner_dob': random_prisoner_dob(),
+                'prison': self.prisons[1].pk
+            },
+        ]
+        response = self.client.post(
+            self.list_url, data=data, format='json'
+        )
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        # test that 2 prisoner location records got created
+        latest_created = PrisonerLocation.objects.filter(upload_counter=2)
+        self.assertEqual(latest_created.count(), len(data))
+        for item in data:
+            self.assertEqual(latest_created.filter(**item).count(), 1)
+
+    def test_create_validation_error(self):
+        user = self.users[0]
+
+        invalid_data_list = [
+            {
+                'data': {},
+                'msg': 'Should fail because invalid format (dict instead of list)'
+            },
+            {
+                'data': [
+                    {}
+                ],
+                'msg': 'Should fail because empty data'
+            },
+            {
+                'data': [
+                    {
+                        'prisoner_number': '*'*1000,
+                        'prisoner_dob': random_prisoner_dob(),
+                        'prison': self.prisons[0].pk
+                    }
+                ],
+                'msg': 'Should fail because empty data'
+            },
+            {
+                'data': [
+                    {
+                        'prisoner_number': random_prisoner_number(),
+                        'prisoner_dob': '01//02//2015',
+                        'prison': self.prisons[0].pk
+                    }
+                ],
+                'msg': 'Should fail because invalid data format'
+            },
+            {
+                'data': [
+                    {
+                        'prisoner_number': random_prisoner_number(),
+                        'prisoner_dob': random_prisoner_dob(),
+                        'prison': 'invalid'
+                    }
+                ],
+                'msg': 'Should fail because invalid prison'
+            },
+        ]
```

Comment:
I still think this should be a bunch of individual tests (maybe using a [data provider](https://pypi.python.org/pypi/unittest-data-provider) or similar), but I'm not bothered enough that it should block us merging.